confirm-deposit

```diff
@@ -1 +1,109 @@
+After five blocks have been created, valid deposits are labeled "confirmed."
+
+Run the javascript below to see pending and confirmed transactions and which block number they were added to.
+
+## One-Time Setup
+
+1. Install [node.js](https://nodejs.org/en/download/). 
+On the command line, run `node -v`  to check your installation.
+
+2. Clone the Flare repository: 
+
+`git clone https://github.com/flare-foundation/flare.git
+cd flare`
+
+See [the Flare repository on GitHub](https://github.com/flare-foundation/flare).
+
+3. Install the web3.js library: 
+
+`npm install web3`
+
+See [Adding web3.js](https://web3js.readthedocs.io).
+
+4. Build Flare with this script:
+
+`./scripts/build.sh`
+
+## Run the Code
+
+1. `cd` to `flare`
+
+2. Run the code: 
+
+The following code:
+
+* Sets up the deposit by instantiating `web3`, assigning the endpoint URL, and assigning the `receivingAddress`
+* Subscribes to 'pending transactions` to listen for deposit transactions and, if they are sent to the receiving address, labels them as "pending."
+* Subscribes to 'newBlockHeaders` to get a block that is five blocks back from the block with the pending transactiions and labels it as "confirmed."
+
+```javascript
+// You must have web3 installed 
+const Web3 = require('web3');
+
+// Test this code on the Coston testnet. Use your own node URL for actual runtime.
+// https://docs.flare.network/dev/reference/coston-testnet/
+const web3 = new Web3("wss://coston-api.flare.network/ext/bc/C/ws");
+
+// This is the testnet receiving address. Use your receiving wallet address for actual runtime.
+const receivingAddress = "0x947c76694491d3fD67a73688003c4d36C8780A97";
+
+// Monitor for newly submitted deposit transactions.
+web3.eth.subscribe("pendingTransactions")
+// When a new transaction hash is received...
+.on("data", async (transactionHash) => {
+    // retrieve the transaction.
+    let tx = await web3.eth.getTransaction(transactionHash);
+    // If it is directed to our address...
+    if (tx.to === receivingAddress) {
+        // mark it as pending.
+        console.log("Transaction", tx.hash, "is pending");
+    }
+}).on("error", console.error);
+
+// Retrieve blocks 5 blocks back for confirmation.
+web3.eth.subscribe("newBlockHeaders")
+// When a new block has been produced...
+.on("data", async (blockHeader) => {
+    // retrieve a block old enough to be considered confirmed.
+    let block = await web3.eth.getBlock(blockHeader.number - 5);
+
+    // Get all of its transactions.
+    block.transactions.forEach(async (transactionHash) => {
+        // Retrieve the transaction...
+        let tx = await web3.eth.getTransaction(transactionHash);
+        // If it is directed to our address...
+        if (tx.to === receivingAddress) {
+            // mark it as confirmed.
+            console.log("Transaction", tx.hash,
+                "is confirmed in block", block.number);
+        }
+    });
+}).on("error", console.error);
+```
+
```

Comment:
Do we have more information on possible error codes, such as Transaction Status](https://github.com/flare-foundation/multi-chain-client/blob/main/docs/definitions/transaction-status.md)

Reply:
Good point. It is always a good idea to talk about return values and error codes. Otherwise, when things do not work users are lost and get very frustrated.
In this case, it would be nice to add that link, but we should check that's actually what's being returned. I am not sure right now.

Reply:
This is from multi-chain client: `Success` doesn't seem applicable, bc we give the "pending" and "confirmed" statuses. However, `failed due to sender` and `failed due to receiver` sound plausible and may be worth looking into.

| ID  | Status                      | Description                                                                                                                                                        |
| --- | --------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `0` | **Success**                 | Transaction was successful and payment was made.                                                                                                                   |
| `1` | **Failure due to sender**   | Transaction failed but it was still included in a block and the fee was paid. The failure cannot be clearly attributed to the receiver.                            |
| `2` | **Failure due to receiver** | Transaction failed but it was still included in a block and the fee was paid. The failure can be clearly attributed to the receiver (e.g. bad destination address) |